api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -897,68 +895,70 @@ public String uploadMappings(MultipartFile mapping) {
 	}
 	
 	@Override
-	public String[] exportSelectedMappings(String[] selectedMappings) {
+	public String[] exportSelectedMappings(String[] selectedMappings) throws IOException {
 		String[] cleanedSelectedMappings = cleanSelectedMappings(selectedMappings);
 		String msg = "";
 		String[] returnStr = new String[2];
 		String path = null;
-		
-		try {
-			byte[] buffer = new byte[1024];
-			String sourceDirectory = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_MAPPINGS_FOLDER
-			        + File.separator;
-			String tempFolderName = OpenmrsUtil.getApplicationDataDirectory() + DHISCONNECTOR_TEMP_FOLDER + File.separator;
-			String suffix = ".mapping.json";
-			String zipFile = tempFolderName + "exported-mappings_" + (new Date()).getTime() + ".zip";
-			
-			(new File(tempFolderName)).mkdirs();
-			
-			FileOutputStream fout = new FileOutputStream(zipFile);
-			ZipOutputStream zout = new ZipOutputStream(fout);
-			File dir = new File(sourceDirectory);
-			
-			if (!dir.isDirectory()) {
-				System.out.println(sourceDirectory + " is not a directory");
-			} else {
-				File[] files = dir.listFiles();
-				String mappings = "";
-				
-				if (files.length == 0) {
-					msg = Context.getMessageSourceService().getMessage("dhisconnector.exportMapping.noMappingsFound");
-				} else {
-					for (int i = 0; i < files.length; i++) {
-						if (files[i].getName().endsWith(suffix)) {
-							FileInputStream fin = new FileInputStream(files[i]);
-							
-							mappings += files[i].getName() + "<:::>";
-							System.out.println("Compressing " + files[i].getName());
-							if (cleanedSelectedMappings.length == 0) {
-								copyToZip(buffer, zout, files, i, fin);
-							} else {
-								if (selectedMappingsIncludes(cleanedSelectedMappings, files[i].getName())) {
-									copyToZip(buffer, zout, files, i, fin);
-								}
-							}
-							msg = Context.getMessageSourceService().getMessage("dhisconnector.exportMapping.success");
-							zout.closeEntry();
-							fin.close();
-						}
-					}
-					if (mappings.split("<:::>").length == 0) {
-						msg = Context.getMessageSourceService().getMessage("dhisconnector.exportMapping.noMappingsFound");
-					}
-					path = zipFile;
-				}
-			}
-			zout.close();
-			System.out.println("Zip file has been created!");
-		}
-		catch (IOException e) {
-			e.printStackTrace();
-		}
-		returnStr[0] = msg;
-		returnStr[1] = path;
-		return returnStr;
+
+		return exportMapping(cleanedSelectedMappings[0]);
+
```

Comment:
@k-joseph 
This is where I call the function.
I'm using the exiting method to get the mapping names and testing the function by importing one of those mappings.

Reply:
i mean, are u calling this in a controller/standalone app?

Reply:
No this is what's happening
We already have an exportSelectedMappings service method on DHIS service file right? The one we use to export mapping.
So I've commented everything inside that function calling this function which I wrote myself, `exportMapping`, inside that function by passing in a mapping name.
And then I navigate into the exiting export mappings UI and trying to export a mapping file, which is calling `exportSelectedMappings` eventually, which will result in calling the `exportMapping`.